### PR TITLE
Improve loading of classes in RunnerClassLoader

### PR DESCRIPTION
**Pull request by geoand**
Improve loading of classes in `RunnerClassLoader`

This ClassLoader is already parallel capable, so just mark it as such.
Furthermore, we define the package without using the ClassLoader lock
by utilizing `getDefinedPackage` and checking for duplicate definitions

### CODE REVIEW

**Comment by geoand** [comment]
This is probably be incorrect because without calling `definePackage` we lose the  "metadata" of the package

**Comment by gsmet** [comment]
My recollection is that we specifically added `definePackage` in some of our class loaders.

**Comment by geoand** [comment]
Yeah... If we need it, I wonder if we can call it without the class loading lock

**Comment by geoand** [comment]
PR updayed

File: independent-projects/bootstrap/runner/src/main/java/io/quarkus/bootstrap/runner/RunnerClassLoader.java [hunk 1]
```diff
@@ -101,18 +105,48 @@ public Class<?> loadClass(String name, boolean resolve) throws ClassNotFoundExce
...
+                        var loaded = getDefinedPackage(pkgName);
+                        if (loaded == null) {
+                            throw e;
+                        }
                     }
-                    throw e;
+                    return;
                 }
             }
+            definePackage(pkgName, null, null, null, null, null, null, null);
```
Comment (dmlloyd) [inline independent-projects/bootstrap/runner/src/main/java/io/quarkus/bootstrap/runner/RunnerClassLoader.java:135]: This one also needs a `try`/`catch` check just like above.

**Review by dmlloyd — COMMENTED** [review]
One comment, otherwise LGTM

File: independent-projects/bootstrap/runner/src/main/java/io/quarkus/bootstrap/runner/RunnerClassLoader.java (hunk 1 above)
Comment (geoand) [reply independent-projects/bootstrap/runner/src/main/java/io/quarkus/bootstrap/runner/RunnerClassLoader.java]: Oops, fixed

**Review by dmlloyd — APPROVED** [review]

**Comment by geoand** [comment]
I'll leave it in draft so I can get a full CI run on my fork before burdening the Quarkus org's (limited) CI

**Comment by geoand** [comment]
@gsmet I forgot to undraft this... Assuming CI go green within the day, is there time to include this in the CR?

**Comment by geoand** [comment]
If it does not make it into CR, I would defer this to the next release

**Comment by gsmet** [comment]
Sorry missed this. CR was already on its way.

**Comment by geoand** [comment]
No problem

**Pull request merged**